test/technology/charge_distribution_surface.cpp

```diff
@@ -0,0 +1,181 @@
+//
+// Created by Jan Drewniok on 05.12.22.
+//
+
+#include <catch2/catch_template_test_macros.hpp>
+#include <fiction/layouts/cartesian_layout.hpp>
+#include <fiction/layouts/cell_level_layout.hpp>
+#include <fiction/layouts/clocked_layout.hpp>
+#include <fiction/layouts/hexagonal_layout.hpp>
+#include <fiction/technology/cell_technologies.hpp>
+#include <fiction/technology/charge_distribution_surface.hpp>
+#include <fiction/technology/sidb_surface.hpp>
+
+using namespace fiction;
+
+TEMPLATE_TEST_CASE(
+    "charge distribution surface traits and construction", "[sidb-surface]",
+    (sidb_surface<cell_level_layout<sidb_technology, clocked_layout<cartesian_layout<offset::ucoord_t>>>>),
+    (sidb_surface<cell_level_layout<sidb_technology, clocked_layout<hexagonal_layout<offset::ucoord_t, odd_row_hex>>>>),
+    (sidb_surface<cell_level_layout<sidb_technology, clocked_layout<hexagonal_layout<offset::ucoord_t, even_row_hex>>>>),
+    (sidb_surface<cell_level_layout<sidb_technology, clocked_layout<hexagonal_layout<offset::ucoord_t, odd_column_hex>>>>),
+    (sidb_surface<cell_level_layout<sidb_technology, clocked_layout<hexagonal_layout<offset::ucoord_t, even_column_hex>>>>))
+{
+    REQUIRE(is_cell_level_layout_v<TestType>);
+    CHECK(has_assign_sidb_defect_v<TestType>);
+    CHECK(has_get_sidb_defect_v<TestType>);
+    CHECK(has_foreach_sidb_defect_v<TestType>);
```

Comment:
It doesn't make much sense to test the defects here. Rather check that the charge state interface is not present.